src/sentry/lang/javascript/processor.py

```diff
@@ -861,19 +861,35 @@ def is_valid_frame(frame):
     return frame is not None and frame.get("lineno") is not None
 
 
-def get_function_for_token(frame, token):
+def get_function_for_token(frame, token, previous_frame=None):
     """
     Get function name for a given frame, based on the looked up token.
-    Return tokens name if we have a usable value from symbolic or we have no initial function name at all,
-    otherwise, fallback to frames current function name.
+    Return tokens name if we have a usable value from symbolic,
+    if possible, ask the callsite whether it's token has a valid name if we
+    did not resolve to anything useful, otherwise, fallback to useless name
+    if we had no function name at all, or to frames current, minified function name.
```

Comment:
This is a bit confusingly worded

Reply:
Totally agree, I had a hard time coming up with something more reasonable 😅

Reply:
Maybe a list of things that will be tried in order could work?

Reply:
Updated